contacts/main.js

```diff
@@ -0,0 +1,253 @@
+    return String.fromCharCode(numbers[rand]);
+  }
+
+  // suffle may be implemented -> TODO
+
+  for (let i=0;i<3;i++){
+    let number = (Math.floor(Math.random() * 10)).toString(10);
+    let char = getRandChar();
+    finalArray.push(number);
+    finalArray.push(char);
+  }
+
+  let arr = '';
+  for (let i=0;i<finalArray.length; i++){
+    arr += finalArray[i];
+  }
+
+  return arr;
+}
+
+function addToHTMLNewPerson(newPerson) {
+  let ul = document.getElementsByClassName('contacts')[0].getElementsByTagName('ul')[0];
+
+  let newLi = document.createElement('li');
+  newLi.className = 'person-field';
+  newLi.id = 'li-' + newPerson.id;
+
+  let plusButton = document.createElement('button');
+  plusButton.id = 'plus-button-' + newPerson.id;
+  plusButton.innerHTML = '+';
+  newLi.appendChild(plusButton);
+
+  let fav_count = document.createElement('div');
+  fav_count.id = 'count-' + newPerson.id;
+  fav_count.className = 'favs-count'
+  fav_count.innerHTML = newPerson.likes;
+  newLi.appendChild(fav_count);
+
+  let minusButton = document.createElement('button');
+  minusButton.id = 'minus-button-' + newPerson.id;
+  minusButton.innerHTML = '-';
+  newLi.appendChild(minusButton);
+
+  let image = document.createElement('img');
+  image.src = newPerson.image;
+  image.alt = 'pic';
+  image.className = 'avatar';
+  newLi.appendChild(image);
+
+  let checkbox = document.createElement('input');
+  checkbox.type = 'checkbox';
+  newLi.appendChild(checkbox);
+
+  let full_name = document.createElement('span');
+  full_name.className = 'full-name';
+  let first_name = document.createElement('span');
+  first_name.innerHTML = newPerson.firstName;
+  first_name.style.marginRight = '10px';
+  full_name.appendChild(first_name);
+  let last_name = document.createElement('span');
+  last_name.innerHTML = newPerson.lastName;
+  full_name.appendChild(last_name);
+  newLi.appendChild(full_name);
+
+  let trash = document.createElement('img');
+  trash.src = 'images/trash-1.png';
+  trash.alt = 'trash';
+  trash.className = 'trash';
+  newLi.appendChild(trash);
+
+  ul.appendChild(newLi);
+}
+
+function attachConections(newPerson){
+  let personHTMLEntity = document.getElementById('li-'+newPerson.id);
+  // for trash
+  personHTMLEntity.getElementsByClassName('trash')[0].onclick = () => {
+    personHTMLEntity.remove();
+    //refresh the favs list
+  }
+  document.getElementById('plus-button-' + newPerson.id).onclick = () => {
+    newPerson.likes+=1;
+    document.getElementById('count-'+newPerson.id).innerHTML = newPerson.likes;
+    //refresh the favs list
+  }
+  document.getElementById('minus-button-' + newPerson.id).onclick = () => {
+    if (newPerson.likes > 0) {
+      newPerson.likes-=1;
+      document.getElementById('count-'+newPerson.id).innerHTML = newPerson.likes;
+    }
+    //refresh the favs list
+  }
+}
+
+function addNewPerson() {
+  // get the data from the form
+  let id = generateUniqueID();
```

Comment:
In javascript there are three keywords for declaring a variable:
1. var
2. let
3. const
Please read the following article to understand them: [var vs. let vs. const](https://tylermcginnis.com/var-let-const/)
Since your variables have immutable values, it's recommended to use const:

```
 const firstName = getFirstNameInput();
 const lastName = getLastNameInput();
 ...
```